ocaml-lsp-server/src/code_actions/action_jump.ml

```diff
@@ -0,0 +1,96 @@
+open Import
+open Fiber.O
+open Stdune
+
+let command_name = "ocamllsp/merlin-jump-to-target"
+
+let targets =
+  [ "fun"; "match"; "let"; "module"; "module-type"; "match-next-case"; "match-prev-case" ]
+;;
+
+let available (capabilities : ShowDocumentClientCapabilities.t option) =
+  match capabilities with
+  | None | Some { support = false } -> false
+  | Some { support = true } -> true
+;;
+
+let error message =
+  Jsonrpc.Response.Error.raise
+  @@ Jsonrpc.Response.Error.make
+       ~code:Jsonrpc.Response.Error.Code.InvalidParams
+       ~message
+       ()
+;;
+
+let command_run server (params : ExecuteCommandParams.t) =
+  let uri, range =
+    match params.arguments with
+    | Some [ json_uri; json_range ] ->
+      let uri = DocumentUri.t_of_yojson json_uri in
+      let range = Range.t_of_yojson json_range in
+      uri, range
+    | None | Some _ -> error "takes a URI and a range as input"
+  in
+  let+ { ShowDocumentResult.success } =
+    let req = ShowDocumentParams.create ~uri ~selection:range ~takeFocus:true () in
+    Server.request server (Server_request.ShowDocumentRequest req)
+  in
+  if not success
+  then (
+    let uri = Uri.to_string uri in
+    Format.eprintf "failed to open %s@." uri);
+  `Null
+;;
+
+(* Dispatch the jump request to Merlin and get the result *)
+let process_jump_request ~merlin ~position ~target =
+  let* results =
+    Document.Merlin.with_pipeline_exn merlin (fun pipeline ->
+      let pposition = Position.logical position in
+      let query = Query_protocol.Jump (target, pposition) in
+      Query_commands.dispatch pipeline query)
+  in
+  match results with
+  | `Error _ -> Fiber.return None
+  | `Found pos -> Fiber.return (Some pos)
+;;
+
+let code_actions
+  (doc : Document.t)
+  (params : CodeActionParams.t)
+  (capabilities : ShowDocumentClientCapabilities.t option)
+  =
+  match Document.kind doc with
+  | `Other -> Fiber.return []
+  | `Merlin merlin ->
+    (match available capabilities with
+     | false -> Fiber.return []
+     | true ->
+       let* actions_fiber_list =
```

Comment:
This value has type `CodeAction.t option list`, a better name would be `actions_list` or `actions` since it  is actually not a fiber.